src/Domain/CoordinateMaps/Wedge3D.cpp

```diff
@@ -41,53 +43,25 @@ operator()(const std::array<T, 3>& x) const noexcept {
-  ReturnType y_inner_cubical_face = eta * radius_of_other_surface_ / sqrt(3.0);
-  // clang-tidy: use auto - This use would be OK, but there used to be
-  // major bugs in this function due to use of auto rather than
-  // ReturnType, so I'd prefer to avoid auto completely here.
-  ReturnType z_inner_cubical_face =  // NOLINT
-      make_with_value<ReturnType>(xi, radius_of_other_surface_ / sqrt(3.0));
-  const ReturnType x_unit_spherical_face = tan(M_PI_4 * xi);
-  const ReturnType y_unit_spherical_face = tan(M_PI_4 * eta);
-  const ReturnType z_spherical_faces_common_factor =
-      make_with_value<ReturnType>(xi, 1.) /
-      sqrt(1.0 + square(x_unit_spherical_face) + square(y_unit_spherical_face));
-  const ReturnType z_inner_spherical_face =
-      radius_of_other_surface_ * z_spherical_faces_common_factor;
-
-  // If sphericity_of_other_surface_==0, the inner_cubical_face variables
-  // correspond to the
-  // face of a cube.
-  // If sphericity_of_other_surface_!=0, the inner_cubical_face variables have
-  // the
-  // spherical_face portions added to them to create a rounded-out cube face.
-  // At sphericity_of_other_surface_==1, the face becomes spherical.
-
-  x_inner_cubical_face +=
-      sphericity_of_other_surface_ *
-      (z_inner_spherical_face * x_unit_spherical_face - x_inner_cubical_face);
-  y_inner_cubical_face +=
-      sphericity_of_other_surface_ *
-      (z_inner_spherical_face * y_unit_spherical_face - y_inner_cubical_face);
-  z_inner_cubical_face += sphericity_of_other_surface_ *
-                          (z_inner_spherical_face - z_inner_cubical_face);
-
-  const ReturnType z_outer_spherical_face =
-      radius_of_spherical_surface_ * z_spherical_faces_common_factor;
-  const ReturnType x_outer_spherical_face =
-      z_outer_spherical_face * x_unit_spherical_face;
-  const ReturnType y_outer_spherical_face =
-      z_outer_spherical_face * y_unit_spherical_face;
-  const ReturnType physical_x =
-      x_inner_cubical_face +
-      (x_outer_spherical_face - x_inner_cubical_face) * (zeta + 1) * 0.5;
-  const ReturnType physical_y =
-      y_inner_cubical_face +
-      (y_outer_spherical_face - y_inner_cubical_face) * (zeta + 1) * 0.5;
-  const ReturnType physical_z =
-      z_inner_cubical_face +
-      (z_outer_spherical_face - z_inner_cubical_face) * (zeta + 1) * 0.5;
+
+  const double r_other = radius_of_other_surface_;
+  const double r_sphere = radius_of_spherical_surface_;
+  const double sphericity = sphericity_of_other_surface_;
+  const ReturnType& blending_factor =
+      (1 - zeta) * sphericity * r_other + (1 + zeta) * r_sphere;
+  const ReturnType& cap_xi = with_equiangular_map_ ? tan(M_PI_4 * xi) : xi;
+  const ReturnType& cap_eta = with_equiangular_map_ ? tan(M_PI_4 * eta) : eta;
+  const ReturnType& one_over_rho =
+      1.0 / sqrt(1.0 + square(cap_xi) + square(cap_eta));
+  const ReturnType& physical_x =
+      0.5 * (1 - zeta) * (1 - sphericity) * (r_other / sqrt(3.0)) * xi +
```

Comment:
reorder math so all doubles are on the left when possible and precompute any common factors